### ISSUE TITLE
Suggestion: Check for nested substitutions (a Zsh-ism)

### ISSUE THREAD

**Issue by 8573** [koalaman/shellcheck#1881]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
# ...
# GitLab CI provides $CI_JOB_IMAGE, which, for us,
# will look like "rust:1.23.45". Strip the "rust:"
# prefix off to get the version number, or let it
# default to "?" if $CI_JOB_IMAGE is undefined.
export rust_version="${${CI_JOB_IMAGE:-?}#rust:}"
```

#### Here's what shellcheck currently says:

```
No issues detected!
```

#### Here's what I wanted or expected to see:

An error message reminding me that nesting variable substitutions in this way is a Zsh-ism that will cause errors in POSIX shell, GNU Bash, etc.

The non-Zsh equivalent would be something such as—

```sh
rust_version="${CI_JOB_IMAGE:-?}"
export rust_version="${rust_version#rust:}"
```
